File: code/443.string-compression.php
<?php
/*
 * @lc app=leetcode id=443 lang=php
 *
 * [443] String Compression
 */

// @lc code=start
class Solution {

    /**
     * @param String[] $chars
     * @return Integer
     */
    function compress(&$chars) {
        $result = [];
        $result[] = $chars[0];
        $count = 1;
        for ($i = 1; $i < count($chars); $i++) {
            if ($chars[$i - 1] === $chars[$i]) {
                $count++;
                continue;
            }

            if ($count > 1) {
                $result = array_merge($result, str_split((string) $count));
            }
            // 新しい文字列
            $count = 1;
            $result[] = $chars[$i];
        }

        // 最後の文字列のカウントを処理
        if ($count > 1) {
            $result = array_merge($result, str_split((string) $count));
        }

        $chars = $result;
        return count($chars);
    }
}
// @lc code=end

File: code/443.string-compression.go
package code

import (
	"strconv"
	"strings"
)

/*
 * @lc app=leetcode id=443 lang=golang
 *
 * [443] String Compression
 */

// @lc code=start
func compress(chars []byte) int {
	result := make([]string, 0)

	for i := 0; i < len(chars); {
		count := 1

		j := i + 1
		for j < len(chars) {
			if chars[j] == chars[i] {
				count++
				j++
			} else {
				break
			}
		}
		result = append(result, string(chars[i]))
		if count != 1 {
			result = append(result, strings.Split(strconv.Itoa(count), "")...)
		}

		i = j
	}

	compressed := strings.Join(result, "")
	for i := 0; i < len(compressed); i++ {
		chars[i] = compressed[i]
	}
	return len(result)
}

// @lc code=end
